mlir-sycl/doc/design/SYCLGridRepresentation.md

```diff
@@ -0,0 +1,244 @@
+# SYCL Index Space Representation
+
+Identifying a work-item in the SYCL index space (using its global id or the
+combination of its work-group id and its local id within a work-group) is
+necessary to perform memory access analysis on SYCL device code, so representing
+these SYCL index space identifiers is required to enable such analysis through
+the SYCL dialect.
+
+**Note**: Through this document, we assume `memref` and `llvm.ptr` are used for
+pointer types, `llvm.struct` for structured types and `sycl.cast` is the
+operation used to perform a casting from a SYCL type to a base class.
+
+## The `sycl.call` operation
+
+The SYCL dialect includes a `sycl.call` operation representing function
+calls. As these SYCL index space identifiers are obtained through member
+function calls, e.g., `sycl::nd_item::get_global_id()`, `sycl.call` is the first
+candidate that comes to mind for this task. However, this operation is too
+low-level to do the work for us. If we look at this example:
+
+```MLIR
+%0 = sycl.call() {FunctionName = @get_global_id, MangledFunctionName = @..., TypeName = @nd_item} : (!memref<?xsycl_nd_item_1_>) -> !sycl_id_1_
+%1 = sycl.call() {FunctionName = @get_local_id, MangledFunctionName = @..., TypeName = @nd_item} : (!memref<?xsycl_nd_item_1_>) -> !sycl_id_1_
```

Comment:
This operations correspond to an implementation using address space 0 for argument pointers. Might correspond to host code.